angrmanagement/ui/widgets/qjobs.py

```diff
@@ -0,0 +1,302 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import qtawesome as qta
+from PySide6.QtCore import QSize, Qt
+from PySide6.QtWidgets import (
+    QAbstractItemView,
+    QHBoxLayout,
+    QHeaderView,
+    QLabel,
+    QProgressBar,
+    QPushButton,
+    QSizePolicy,
+    QTableWidget,
+    QTableWidgetItem,
+    QVBoxLayout,
+    QWidget,
+)
+
+if TYPE_CHECKING:
+    from angrmanagement.data.jobs import job
+
+
+class CancelButton(QPushButton):
+    """
+    This creates a cancel button for each job in the job views, using the job.cancel()
+    method or skipping it from the queue
+    """
+
+    def __init__(self, table: QJobs, new_job: job, workspace):
+        super().__init__("Cancel")
+        self.workspace = workspace
+
+        self.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
+
+        self.clicked.connect(lambda: self.onClick(table, new_job))
+
+    # On cancel button click, the function will check whether to cancel or skip the job
+    def onClick(self, table, the_job):
+        # Job is cancelled/interrupted if running
+        if isinstance(the_job.status, RunningWidget) and the_job in self.workspace.job_manager.jobs:
+            self.workspace.job_manager.worker_thread.keyboard_interrupt()
+            the_job.cancelled = True
+            table.change_job_cancel(the_job)
+        # Job is skipped if it's pending
+        elif isinstance(the_job.status, PendingWidget):
+            the_job.cancelled = True
+            table.change_job_cancel(the_job)
+
+
+class CancelWidget(QWidget):
+    """
+    This creates a status widget to show a job's status that it is
+    cancelled
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
+
+        # Constructing cancel icon widget
+        self.cancel_widget = qta.IconWidget()
+        cancel_icon = qta.icon("ei.remove-sign", color="red")
+        self.cancel_widget.setIconSize(QSize(45, 26))
+        self.cancel_widget.setIcon(cancel_icon)
+
+        # Create a horizontal layout and add the labels
+        hbox = QHBoxLayout()
+        hbox.addWidget(self.cancel_widget)
+
+        # Set the layout for the widget
+        self.setLayout(hbox)
+
+
+class FinishWidget(QWidget):
```

Comment:
Did you consider using a single widget for the icon and updating it instead of having separate widgets for each? In any case, maybe this should be `FinishedWidget`, since it reflects the state rather than an action?